Mustache/src/js/main.js

```diff
@@ -0,0 +1,96 @@
+//const Mustache = require('mustache');
+
+var data = {};
+
+function validar(e) {
+    tecla = (document.all) ? e.keyCode : e.which;
+    if (tecla===13) callPromise();
+}
+
+function promiseC() {
+    return new Promise(function(resolve, reject) {
+    var request = new XMLHttpRequest();
+    request.open('GET', "",true);
+    request.onload = function() {
+        if (request.status === 200) {
+        resolve(request.response);
+        } else {
+        reject(Error("Error"));
+        }
+    };
+    request.onerror = function() {
+        reject(Error('There was a network error.'));
+    };
+    request.send();
+    });
+}
+
+function callPromise(){
+    var userInput = document.getElementById("input").value;
+    promiseC().then(function(response) {
+        showEspecificInfo(userInput);
+    }, function(Error) {
+        console.log(Error);
+        show(data);
+    });
+}
+
+   
+
+function showEspecificInfo(userInput){
+    if (userInput !== "") {
+        var companieJson = getAgentByName(userInput);
+        if(companieJson !== undefined){
+            var lis = [companieJson];
+            var s = {"companies":[companieJson]};
+            console.log("n " + s);
+            show(s);
+            return;
+        }
+    }
+    
+}
+
+function getAgentByName(name) {
+    var jsonCompañies = data;
+    for (let index = 0; index < jsonCompañies["companies"].length; index++) {
+        const element = jsonCompañies["companies"][index];
+        if(element["name"] === name){
+            return jsonCompañies["companies"][index];
+        }
+    }
+}
+
+function showAll(){
+    show(data);
+}
+
+function show(dataJ) {
+    const template = fetch('../mustache/template.mustache').then(response => response.text());
```

Comment:
Just if you didn't notice, the line `response => response.text()` is an arrow function, instead of writing `.then(function(response) { return response.text()})`. Better, shorter and has some other advantages as well.